pandas/io/formats/format.py

```diff
@@ -932,12 +938,17 @@ def __init__(self, values, digits=7, formatter=None, na_rep='NaN',
         self.quoting = quoting
         self.fixed_width = fixed_width
         self.leading_space = leading_space
+        self.shortcut = shortcut
 
     def get_result(self):
         fmt_values = self._format_strings()
         return _make_fixed_width(fmt_values, self.justify)
 
     def _format_strings(self):
+        # shortcut
+        if self.formatter is not None and self.shortcut:
```

Comment:
why do you need shortcut at all? isn't ``self.formatter is not None`` enough?

Reply:
the implementation of ExtensionArrayFormatter relies a formatter parameter and not shortcutting. so yes we need a shortcut parameter from to_string etc unless the ExtensionArrayFormatter implementation is changed.

Reply:
I would rather do that, this is too hacky

Reply:
changing the ExtensionArrayFormatter is unlikely to be trivial.